docs/user/exercises/programming-exercise-AI-proof-guidelines.inc

```diff
@@ -0,0 +1,487 @@
+      as a basic concept for understanding linked lists. The task of this exercise
+      were to implement the common List functionalities from scratch rather than
+      rely on the standard methods from java. ListNode and LinkedList were
+      therefore own classes that needed to be filled with the required
+      functionalities.
+
+
+   This depiction provides an understanding of how a linked list should look
+   like after the add function was called.
+
+   .. figure:: programming/AI-proof-exercises/add.png
+      :align: center
+      :width: 50%
+
+
+      Figure shows what a list should look like after the add functionality was called. Picture from the Problem statement
+
+-  **Java frameworks:** Using Java frameworks, such as JavaFX, is challenging for AI
+   and helps to visualize projects in a more interactive way
+   for students through gamification. How can we incorporate that for
+   our exercises?
+
+   -  Provide Graphical User Interfaces (GUI) for
+      students with which they can also test the functionality of their
+      code.
+
+   -  Give exercises that use GUIs because AI has difficulty
+      solving them accurately.
+
+   -  Provide pictures of how the GUI should look in the end.
+
+      Students needed to implement the GUI based on the this figure that
+      was given in the problem statement.
+
+      .. figure:: programming/AI-proof-exercises/GUI-upgrade-for-game.png
+         :align: center
+         :width: 30%
+
+
+         Example GUI from homework exercise
+
+   -  Providing GIFs is also possible and shows a good way to show the dynamic behaviour of the GUI.
+      This also ensures less likelyhood to copy this format into various AI tools.
+
+
+.. note::
+   The unpaid version of ChatGPT is limited to text input like many other but not all LLMs. Including a fair amount of graphical
+   representation for the exercises helps to limit the effectiveness of such tools.
+   Even though there are other LLMs that can also use graphical representation, mixing up the representation with text and graphics
+   is always a good idea.
+
+Problem statement and exercise instructions
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+
+-  **Exercise length:** At this point of time exercise length has an effect on the performance of AI tools and can be considered for
+   certain exercises. This will change drastically as LLMs get more advanced.
+
+   -  Studies suggest, exercises between 500 and 1000 characters are easier for AI tools,
+      while 1200 and 2500 characters lead to worse performance
+      Short exercises are normally easier to solve for AI since there are not so many
+      constraints and the problem statement only focused on the most
+      relevant implementation parts.
+
+   .. note::
+       Apply this with more complex exercises at a later stage of the semester. In the first exercises
+       it does not make sense to give more text just to make the exercise more complicated.
```

Comment:
I feel like making exercise statements so hard that AI can't understand them while humans can is a hard thing to do. I don't think making exercise statements longer for the sake of them being longer isn't the best strategy at any point in the semester.